api/swagger.yaml

```diff
@@ -157,6 +157,293 @@ tags:
+              Name: "json-file"
+              Options:
+                max-file: "3"
+                max-size: "10M"
+            Placement: {}
+            Resources:
+              Limits:
+                MemoryBytes: 104857600
+              Reservations: {}
+            RestartPolicy:
+              Condition: "on-failure"
+              Delay: 10000000000
+              MaxAttempts: 10
+          Mode:
+            Replicated:
+              Replicas: 4
+          UpdateConfig:
+            Parallelism: 2
+            Delay: 1000000000
+            FailureAction: "pause"
+            Monitor: 15000000000
+            MaxFailureRatio: 0.15
+          RollbackConfig:
+            Parallelism: 1
+            Delay: 1000000000
+            FailureAction: "pause"
+            Monitor: 15000000000
+            MaxFailureRatio: 0.15
+          EndpointSpec:
+            Ports:
+              -
+                Protocol: "tcp"
+                PublishedPort: 8080
+                TargetPort: 80
+          Labels:
+            foo: "bar"
+  SecretCreateBody:
+    allOf:
+      - $ref: "#/definitions/SecretSpec"
+      - type: "object"
+        example:
+          Name: "app-key.crt"
+          Labels:
+            foo: "bar"
+          Data: "VEhJUyBJUyBOT1QgQSBSRUFMIENFUlRJRklDQVRFCg=="
+          Driver:
+            Name: "secret-bucket"
+            Options:
+              OptionA: "value for driver option A"
+              OptionB: "value for driver option B"
+  ConfigCreateBody:
+    allOf:
+      - $ref: "#/definitions/ConfigSpec"
+      - type: "object"
+        example:
+          Name: "server.conf"
+          Labels:
+            foo: "bar"
+          Data: "VEhJUyBJUyBOT1QgQSBSRUFMIENFUlRJRklDQVRFCg=="
+  ContainerUpdateUpdate:
+    allOf:
+      - $ref: "#/definitions/Resources"
+      - type: "object"
+        properties:
+          RestartPolicy:
+            $ref: "#/definitions/RestartPolicy"
+    example:
+      BlkioWeight: 300
+      CpuShares: 512
+      CpuPeriod: 100000
+      CpuQuota: 50000
+      CpuRealtimePeriod: 1000000
+      CpuRealtimeRuntime: 10000
+      CpusetCpus: "0,1"
+      CpusetMems: "0"
+      Memory: 314572800
+      MemorySwap: 514288000
+      MemoryReservation: 209715200
+      KernelMemory: 52428800
+      RestartPolicy:
+        MaximumRetryCount: 4
+        Name: "on-failure"
+  ContainerCreateBody:
+    allOf:
+      - $ref: "#/definitions/ContainerConfig"
+      - type: "object"
+        properties:
+          HostConfig:
+            $ref: "#/definitions/HostConfig"
+          NetworkingConfig:
+            description: "This container's networking configuration."
+            type: "object"
+            properties:
+              EndpointsConfig:
+                description: "A mapping of network name to endpoint configuration for that network."
+                type: "object"
+                additionalProperties:
+                  $ref: "#/definitions/EndpointSettings"
+    example:
```

Comment:
In recent changes to the swagger, I tried to keep the example values per-property, instead of providing a full example. I've seen that various example objects don't match the actual structure of the definition. As a result, the API documentation only shows the provided example as-is, which can be misleading.

(Also see my other comment - we should try to not duplicate definitions if not possible)

Reply:
@thaJeztah So if I understand correctly you want `allOf` eliminated where unnecessary, and then the associated `example` inlined to the property level where `allOf` has been eliminated.

Reply:
Yes; having the examples directly in the properties of the definitions makes it easier to maintain (i.e., every place where the definition is used, those example values are automatically used).

Also, it prevents incorrect/incomplete example responses to be written for an endpoint (see my other comment)

Reply:
So something like 4317d25c43a628e946fe6e9d8bb710a48b93bfb4, yes? There are still some inline `allOf`s that do add extra properties (e.g. `ContainerCreateBody`), so I'm going to leave those as extracted definitions, but inline the examples.

Reply:
Also, when I'm inlining examples like this, do you want me to discard conflicting examples? The problem is that there might be certain combinations of properties that are valid together in a way that cannot be specified using the schema validation, and can only be illustrated by example. 

E.g. the `ServiceSpec` instance provided as an example for `/services/create` may have a meaningfully different set of properties than the one for `/services/{id}/update`. If I just inline this to the property level, any such meaningful distinction is lost.

Reply:
Sorry for the delay: I think it's ok to remove those full examples; I don't think they have been tested, and most values are deliberately made "example-ish" to show they're just example values.

Perhaps it would be good to mention that in the introduction section of the documentation.

Reply:
@thaJeztah Is there any chance you'd accept the PR as-is, sans example inlining? Inlining these examples is pretty tedious, and isn't really necessary for fixing the bug the current spec is suffering from. I'm trying to use this spec for automatically generating a [JS client](https://www.npmjs.com/package/docker-client) for the Docker API. Currently I'm pointing to my personal fork, which has this PR merged, because the client doesn't generate otherwise.

I'd automate the inlining, but can't do that because the current spec isn't formatted, and it would touch basically every line in the spec if I parsed, modified, and reserialized it. If you're fine with that I can try doing it that way.

Reply:
Although perhaps I'm misunderstanding you. When you say "I think it's ok to remove those full examples", are you saying just delete them without replacing them somewhere else?

Reply:
ping @thaJeztah

Reply:
@cpuguy83 Who would be the best person to take a look at this? There's open issues in swagger codegen that can't be resolved until this bug in Docker is fixed.

Reply:
@thaJeztah any chance you can take another look at this? I opened one of those issues in `swagger-codegen` and would love to see this PR make progress.